Failed to add chart repository with helm when OIDC is enabled (bug still active)

following up on #7625 

**Expected behavior and actual behavior:**
Helm push should work with OIDC Cli Secret

**Steps to reproduce the problem:**
helm push redmine-9.1.0.tgz https://mru:4b01foduqa.....@registry.company/chartrepo/library

**Versions:**
Please specify the versions of following systems.
- harbor version: [1.8.1]
- docker engine version: [18.09.3]
- docker-compose version: [1.24.0]

**Additional context:**
Docker login with cli secret works
```
echo 4b01foduqa..... | docker login --username mru --password-stdin https://registry.company
Login Succeeded
```
helm push with cli secret won't work
```
helm push redmine-9.1.0.tgz https://mr:4b01foduqa.....@registry.company/chartrepo/library
Error: 401: could not properly parse response JSON: {"code":401,"message":"Unauthorized"}
[ERROR] [/core/filter/security.go:471]: failed to authenticate mru: Unrecognized auth_mode: oidc_auth
[ERROR] [/common/api/base.go:69]: POST /api/chartrepo/library/charts failed with error: {"code":401,"message":"Unauthorized"}
```

Anything else to do with Helm push here? I've tried using helm push following the manual (https://github.com/goharbor/harbor/blob/master/docs/user_guide.md#working-with-helm-cli), but no matter what method : always 401

--- a/src/core/filter/security.go
+++ b/src/core/filter/security.go
@@ -229,8 +229,10 @@ type oidcCliReqCtxModifier struct{}
 
 func (oc *oidcCliReqCtxModifier) Modify(ctx *beegoctx.Context) bool {
 	path := ctx.Request.URL.Path
-	if path != "/service/token" && !strings.HasPrefix(path, "/chartrepo/") {
-		log.Debug("OIDC CLI modifer only handles request by docker CLI or helm CLI")
+	if path != "/service/token" &&
+		!strings.HasPrefix(path, "/chartrepo/") &&
+		!strings.HasPrefix(path, "/api/chartrepo/") {
+		log.Debug("OIDC CLI modifier only handles request by docker CLI or helm CLI")
 		return false
 	}
 	if ctx.Request.Context().Value(AuthModeKey).(string) != common.OIDCAuth {

